### PR TITLE
Use IDs instead of names in table key generation.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

File: client/table.go [hunk 1]
```diff
@@ -40,30 +40,43 @@ import (
...
-// - Normalize column names to lowercase.
-//
 // - Allow usage of `map[string]interface{}` in place of a struct. Probably
 //   need table schemas first so we know which columns exist.
 //
 // - Add support for namespaces. Currently namespace ID 0 is hardcoded.
 
+var (
+	// TODO(pmattis): Use the appropriate constant in keys.go when it appears.
+	tableDataPrefix = []byte("tbl-")
```
Comment (spencerkimball) [inline client/table.go]: Why do we need anything here? If the access to cockroach is limited to tables...

File: client/table.go [hunk 2]
```diff
@@ -40,30 +40,43 @@ import (
...
 type model struct {
-	name         string   // Table name.
-	fields       fieldMap // The fields of the model type.
-	primaryKey   []string // The columns that compose the primary key.
-	otherColumns []string // All non-primary key columns.
+	name          string // Table name.
+	desc          *proto.TableDescriptor
+	columnsByName map[string]*column
+	columnsByID   map[uint32]*column
+	primaryKey    []*column // The columns that compose the primary key.
```
Comment (spencerkimball) [inline client/table.go]: Kind of strange to use `*column` here and string for `otherColumns`

File: client/table.go [hunk 3]
```diff
@@ -40,30 +40,43 @@ import (
...
+)
+
+func lowerStrings(s []string) []string {
+	for i := range s {
+		s[i] = strings.ToLower(s[i])
+	}
+	return s
+}
+
 // model holds information about a particular type that has been bound to a
```
Comment (tamird) [inline client/table.go]: this comment got reparented

**Comment by spencerkimball** [comment]
LGTM

File: client/table.go (hunk 1 above)
Comment (petermattis) [inline client/table.go]: The `"tbl-"` prefix is intended to be temporary during the transition from key-value to table-based API.

File: client/table.go (hunk 3 above)
Comment (petermattis) [inline client/table.go]: Oops.

File: client/table.go (hunk 2 above)
Comment (petermattis) [inline client/table.go]: Renaming `otherColumns` to `otherColumnNames`.

**Pull request merged**